protokt-codegen/src/main/kotlin/protokt/v1/codegen/generate/Implements.kt

```diff
@@ -77,7 +77,7 @@ internal object Implements {
                     field.name,
                     (field.returnType.classifier as KClass<*>).asTypeName()
                         .let {
-                            if (assumeNotNull) {
+                            if (assumeNotNull && !field.returnType.isMarkedNullable) {
```

Comment:
Can't delegate a field and mark is as non-nullable if the delegated field is nullable in the interface.